astropy/modeling/functional_models.py

```diff
@@ -714,20 +749,33 @@ class Sine1D(Fittable1DModel):
     @staticmethod
     def evaluate(x, amplitude, frequency, phase):
         """One dimensional Sine model function"""
-
-        return amplitude * np.sin(2 * np.pi * frequency * x + 2 * np.pi * phase)
+        argument = TWOPI * (frequency * x + phase)
+        if isinstance(argument, Quantity):
```

Comment:
I wonder about this too. The code works but I'm not sure I understand why it's done this way.